example/canny.cpp

```diff
@@ -0,0 +1,165 @@
+#include <boost/gil.hpp>
+#include <boost/gil/extension/io/jpeg.hpp>
+#include <boost/gil/extension/numeric/sampler.hpp>
+#include <boost/gil/extension/numeric/resample.hpp>
+#include <boost/gil/extension/numeric/kernel.hpp>
+#include <boost/gil/extension/numeric/convolve.hpp>
+
+#include <math.h>
+
+using namespace std;
+using namespace boost::gil;
+
+//Example how to detect edges via Canny algorithm
+
+template <typename SrcView>
+void smoothing(const SrcView &src)
+{
+    float gaussian[] = {0.06136f, 0.24477f, 0.38774f, 0.24477f, 0.06136f};
```

Comment:
Since this is an example, it would be very useful to explain these, add comment about source and purpose of the constants, i.e. 1-D gaussian kernel, size of kernel, what it is used for.